.github/actions/update-charm-pins/main.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 Canonical Ltd.
+
+"""Updates pinned versions of charms in tests."""
+
+import logging
+import os
+import sys
+
+from httpx import Client
+from ruamel.yaml import YAML
```

Comment:
Are you using ruamel.yaml rather than pyyaml (which ops itself uses) from personal preference, or does it have functionality that's used here that pyyaml is missing? This is only a CI utility script, but in general keeping the number of packages down is good.

Reply:
1. for (limited) round-trip safety
2. to insert comments

Reply:
Seems reasonable. Perhaps at some point we should consider using it for ops as well? It's a little icky using two packages that do (more-or-less) the same thing, and if this is better, it might be worth it. MPL2 might be an issue in ops itself, though,

Reply:
True, though it seems that 2-packages approach is kindof the way forward.

See e.g. https://peps.python.org/pep-0680/#including-an-api-for-writing-toml